File: min_element.m
function minelts = min_element(list, ordering)

infinities = find(list==inf, 1);
if ~isempty(infinities)
    minelts = inf;
    return
end

n = size(ordering, 1);

killer = zeros(1,n);
killer(list) = 1;

killed = ordering'*killer';
minelts = find(~killed);

minelts = intersect(minelts, list);

end

File: python/min_element.py

import numpy as np
import scipy
import matcompat

# if available import pylab (from matlibplot)
try:
    import matplotlib.pylab as plt
except ImportError:
    pass

def min_element(list, ordering):

    # Local Variables: minelts, ordering, infinities, killer, list, n, killed
    # Function calls: min_element, zeros, intersect, isempty, inf, find, size
    infinities = nonzero((list == np.inf), 1.)
    if not isempty(infinities):
        minelts = np.inf
        return []
    
    
    n = matcompat.size(ordering, 1.)
    killer = np.zeros(1., n)
    killer[int(list)-1] = 1.
    killed = np.dot(ordering.conj().T, killer.conj().T)
    minelts = nonzero((not killed))
    minelts = intersect(minelts, list)
    return [minelts]